shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -643,8 +643,21 @@ - (void)dispatchMouseEvent:(NSEvent*)event phase:(FlutterPointerPhase)phase {
       pixelsPerLine = 40.0;
     }
     double scaleFactor = self.flutterView.layer.contentsScale;
-    flutterEvent.scroll_delta_x = -event.scrollingDeltaX * pixelsPerLine * scaleFactor;
-    flutterEvent.scroll_delta_y = -event.scrollingDeltaY * pixelsPerLine * scaleFactor;
+    // When mouse input is received while shift is pressed (regardless of
+    // any other pressed keys), Mac automatically flips the axis. Other
+    // platforms do not do this, so we flip it back to normalize the input
+    // received by the framework. The keyboard+mouse-scroll mechanism is exposed
+    // in the ScrollBehavior of the framework so developers can customize the
+    // behavior.
+    // At time of change, Apple does not expose any other type of API or signal
+    // that they X/Y axes have been flipped.
+    if (event.modifierFlags & NSShiftKeyMask) {
+      flutterEvent.scroll_delta_x = -event.scrollingDeltaY * pixelsPerLine * scaleFactor;
+      flutterEvent.scroll_delta_y = -event.scrollingDeltaX * pixelsPerLine * scaleFactor;
+    } else {
+      flutterEvent.scroll_delta_x = -event.scrollingDeltaX * pixelsPerLine * scaleFactor;
+      flutterEvent.scroll_delta_y = -event.scrollingDeltaY * pixelsPerLine * scaleFactor;
+    }
```

Comment:
It would be easier to see what the differences are here, and avoid accidental drift in future changes, if the calculations were extracted to locals and then just assigned differently:

```
double scaledDeltaX = -event.scrollingDeltaX * pixelsPerLine * scaleFactor;
double scaledDeltaY = -event.scrollingDeltaY * pixelsPerLine * scaleFactor;
if (event.modifierFlags & NSShiftKeyMask) {
  flutterEvent.scroll_delta_x = scaledDeltaY;
  flutterEvent.scroll_delta_y = scaledDeltaX;
} else {
  flutterEvent.scroll_delta_x = scaledDeltaX;
  flutterEvent.scroll_delta_y = scaledDeltaY;
}
```